apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLoggerFactoryExtensions.cs

```diff
@@ -68,16 +67,7 @@ public static class GoogleLoggerFactoryExtensions
         public static ILoggerFactory AddGoogle(this ILoggerFactory factory, LogTarget logTarget,
             LoggerOptions options = null, LoggingServiceV2Client client = null)
         {
-            // Check params and set defaults if unset.
-            GaxPreconditions.CheckNotNull(factory, nameof(factory));
-            GaxPreconditions.CheckNotNull(logTarget, nameof(logTarget));
-            client = client ?? LoggingServiceV2Client.Create();
-            options = options ?? LoggerOptions.Create();
-
-            // Get the proper consumer from the options and add a logger provider.
-            IConsumer<LogEntry> consumer = LogConsumer.Create(client, options.BufferOptions, options.RetryOptions);
-            GoogleLoggerProvider provider = new GoogleLoggerProvider(consumer, logTarget, options);
-            factory.AddProvider(provider);
+            factory.AddProvider(GoogleLoggerProvider.Create(logTarget, options, client));
```

Comment:
Don't we still need the precondition for factory being non-null?

Reply:
Yes, fixed.